### PR TITLE
Evaluator Base Interface

### PR DESCRIPTION
**Pull request by ankit-oss**
Defined base class interfaces for building evaluator as a generic implementation for black box DP implementations to leverage

Created a sample implementation of BlackBoxPrivacy Interface and tested it

Sample implementation of a DP library that releases statistics by implementing BlackBoxPrivacyInterface. It takes in the privacy and evaluation params for its analysis and releases noisy responses for evaluator to act on.

**Test Result**
(oss_dp) ankit@ANKIT-DELL:~/privacytoolsproject/whitenoise-system/tests$ pytest -k "test_dp_blackbox"
================================ test session starts ===========================
platform linux -- Python 3.8.0, pytest-5.3.1, py-1.8.0, pluggy-0.13.1
rootdir: /mnt/c/Users/ankitsri/source/code/privacytoolsproject/whitenoise-system/tests, inifile: pytest.ini
collected 116 items / 115 deselected / 1 selected                                                                                         

sdk/evaluation/test_interface.py .                                                                                                  [100%]

=========================== 1 passed, 115 deselected in 3.41s ========================= 

Related #216

### CODE REVIEW

File: sdk/opendp/whitenoise/evaluation/blackbox/_base.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+from opendp.whitenoise.evaluation.params._privacy_params import PrivacyParams
+from opendp.whitenoise.evaluation.params._eval_params import EvaluatorParams
+from opendp.whitenoise.evaluation.report._report import Report
+from abc import ABC, abstractmethod
+
+class BlackBoxPrivacyInterface(ABC):
```
Comment (eedeleon) [inline sdk/opendp/whitenoise/evaluation/blackbox/_base.py:6]: why is this internal?

Comment (eedeleon) [reply sdk/opendp/whitenoise/evaluation/blackbox/_base.py:6]: nvm, all of it is so we can test it

Comment (ankit-oss) [reply sdk/opendp/whitenoise/evaluation/blackbox/_base.py:6]: Ya, I really liked the internal convention which allows me to check-in and then we can make it external when I implement the interfaces and add unit tests that pass. So, for skeleton, keeping it internal for now.

File: sdk/opendp/whitenoise/evaluation/report/_report.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import pandas as pd
+import numpy as np
+
+class Report:
```
Comment (eedeleon) [inline sdk/opendp/whitenoise/evaluation/report/_report.py:4]: How does this differ from the core library report?

Comment (ankit-oss) [reply sdk/opendp/whitenoise/evaluation/report/_report.py:4]: That's a good question. This one is agnostic to any specific DP implementation. All it needs is to pass the result of a PrivateQuery or DP function as a dataframe that we can then use for doing our evaluator checks. So quite lightweight. It allows for multiple dimensions (i.e. group by fields) and numerical columns within output to be tested in one-go.

**Comment by ankit-oss** [comment]
Thanks Eddie!

**Pull request merged**